lib/engine/game/g_2038/game.rb

```diff
@@ -652,6 +652,52 @@ def game_hexes
           GAME_HEXES
         end
 
+        EVENTS_TEXT = Base::EVENTS_TEXT.merge(
+          'group_b_corps_available' => ['Group B Corporations become available'],
+          'group_c_corps_available' => ['Group C Corporations become available'],
+        ).freeze
+
+        def setup
+          @al_corporation = corporation_by_id('AL')
+          @al_corporation.capitalization = :incremental
+
+          @corporations.reject! { |c| c.id == 'AL' }
+
+          return if optional_variant_start_pack
+
+          @available_corp_group = :groupA
+
+          @corporations, @b_group_corporations = @corporations.partition do |corporation|
+            corporation.type == :groupA
+          end
+
+          @b_group_corporations, @c_group_corporations = @b_group_corporations.partition do |corporation|
+            corporation.type == :groupB
+          end
+        end
+
+        def event_group_b_corps_available!
+          @log << 'Group B corporations are now available'
+
+          @corporations += @b_group_corporations
```

Comment:
@corporations <<

Reply:
@b_group_corporations is an array, I wouldn't want to append that, not insert it, right?

Reply:
ok, then do concat